web-app/src/lib/db/services/agent.service.ts

```diff
@@ -78,3 +81,53 @@ export async function getHiredAgentsOrderedByLatestJob(userId: string) {
     return bLatestJob.startedAt.getTime() - aLatestJob.startedAt.getTime();
   });
 }
+
+export async function getAgentInputSchema(agentId: string) {
+  const agent = await getAgentById(agentId);
+
+  if (!agent) {
+    throw new Error(`Agent with ID ${agentId} not found`);
+  }
+
+  const agentUrl = agent.apiBaseUrl;
+
+  if (!agentUrl) {
+    throw new Error(`Agent with ID ${agentId} has no API base URL`);
+  }
+
+  let agentUrlString = agentUrl.toString();
+  if (agentUrlString.endsWith("/")) {
+    agentUrlString = agentUrlString.slice(0, -1);
+  }
+
+  const response = await fetch(`${agentUrlString}/input_schema`);
```

Comment:
Has it a reason that you do not use URL Constructor to build the url?

If we use `URL` we don't have to check for an ending `/` in the baseURL.

```typescript
const baseUrl = 'https://example.com/';
const path = '/api/data';
const params = new URLSearchParams({
  userId: '123',
  type: 'full'
});

const url = new URL(path, baseUrl);
url.search = params.toString();

console.log(url.toString()); // Outputs: https://example.com/api/data?userId=123&type=full
```

Reply:
Resolved https://github.com/masumi-network/sokosumi/pull/208/commits/c737666945b528fe98185b993b49143e90821578